src/Stratis.SCL/ECRecover.cs

```diff
@@ -37,6 +37,35 @@ public static bool TryGetSigner(byte[] message, byte[] signature, out Address ad
             }
         }
 
+        /// <summary>
+        /// Retrieves the address of the signer of an ECDSA signature.
+        /// </summary>
+        /// <param name="message">The message that was signed.</param>
+        /// <param name="signature">The ECDSA signature prepended with header information specifying the correct value of recId.</param>
+        /// <param name="address">The Address for the signer of a signature.</param>
+        /// <returns>A bool representing whether or not the signer was retrieved successfully.</returns>
+        public static bool TryGetSignerNoHash(byte[] message, byte[] signature, out Address address)
+        {
+            address = Address.Zero;
+
+            if (message == null || signature == null)
+                return false;
+
+            // NBitcoin is very throwy
+            try
+            {
+                PubKey pubKey = PubKey.RecoverFromMessage(message, Convert.ToBase64String(signature));
```

Comment:
It is important to note that the hash algorithm used here to hash the message under the hood is SHA-256 instead of Keccak256 (which is used in the `TryGetSigner` method). Depending on how the signature was obtained this may be a significant difference, so it should at least be documented to make it obvious. Perhaps the method name should change.

I presume Keccak256 was chosen for similarity to `ecrecover` in Solidity, whereas message signing in the rest of our full node follows the Bitcoin paradigm of SHA-256. Changing the latter is undesirable and changing the former might make interoperability more difficult, although I'm not sure how many current use cases utilise the SCL other than SSAS.

Reply:
This method is specifically not using hashing due to the way that SignMessage API signs a message in the SSAS context.

Reply:
I am referring to the hashing performed inside `RecoverFromMessage`

Reply:
So perhaps we could rename this to `TryGetSignerSHA256` then?

Reply:
That seems reasonable to me

Reply:
Done.